### ISSUE TITLE
SunSpreadAngle must be capitalized

### ISSUE THREAD

**Issue by ozxybox** [TeamSpen210/HammerAddons#274]
It turns out, this keyvalue, `SunSpreadAngle`, must always be properly capitalized for it to be read by VRAD. By having it lowercase in the FGD, the final result will be a lowercased keyvalue in the entity lump, which VRAD skips over.

https://github.com/TeamSpen210/HammerAddons/blob/47afc57e3ff272ebe75a8277b36514bb0a2f10ef/fgd/point/light/light_environment.fgd#L16

The variable is read from here:
https://github.com/ValveSoftware/source-sdk-2013/blob/0d8dceea4310fde5706b3ce1c70609d72a38efdf/mp/src/utils/vrad/lightmap.cpp#L1483C1-L1483C64

The case-sensitive `strcmp` that checks it is here:
https://github.com/ValveSoftware/source-sdk-2013/blob/0d8dceea4310fde5706b3ce1c70609d72a38efdf/mp/src/utils/vrad/lightmap.cpp#L1470C1-L1470C31

**Comment by TeamSpen210** [comment]
That's quite dumb. I think I'll also add a postcompiler transform to adjust the casing as well then, to fix this issue.

**Comment by ozxybox** [comment]
Related, does HA always emit (or intake when importing) lowercase keyvalue names?

**Comment by TeamSpen210** [comment]
Nah, `srctools` does lookups case-insensitively, but preserves the casing used when it’s initially set/parsed. I need to revamp this though, currently it’s directly exposed as the `keys` dict meaning it has to just loop over all the keys each time…

**Comment by ozxybox** [comment]
Ah okay, thanks! Was wondering as I was futzing around with it the other day
